src/client/app/shared/form-input/radiobuttons-input.component.ts

```
@@ -0,0 +1,63 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { AbstractInput } from './abstract-input.component';
+
+export interface RadioButtonOption {
+    label: string;
+    value: string;
+}
+
+@Component({
+    moduleId: module.id,
+    selector: 'radiobuttons-input',
+    templateUrl: 'radiobuttons-input.component.html',
+    styleUrls: ['form-input.component.css']
+})
+export class RadioButtonsInputComponent extends AbstractInput implements OnInit {
+    @Input() options: RadioButtonOption[] = [];
+    public error: string = '';
+
+    constructor() {
+        super();
+    }
+
+   /**
+    * Initialize relevant variables when the directive is instantiated
+    */
+    ngOnInit() {
+        super.ngOnInit();
+        this.parseInputOptions();
+        this.addValidatorsToFormControl();
+    }
+
+    protected addValidatorsToFormControl() {
+        let validators: any = [];
+        if (this.required) {
+            validators.push(Validators.required);
+        }
+        setTimeout( () => {
+            this.formControl.setValidators(validators);
+        });
+    }
+
+    protected updateChanges() {
+        this.formControl.markAsDirty();
+    }
+
+    protected parseInputOptions(): void {
+        if (!this.formControl.value) {
```

Can this happen? Is there a default value?

---

This is to check if one of the radio buttons is toggled on (no one will be toggled on when the model value is null), also to ensure the model value is one of the option's values, for example, the model value for siteStatus radio buttons must be either "PUBLIC" or "PRIVATE". 

Yes, we can set "PUBLIC" as the default value for siteStatus radio buttons, but this may hide some potential errors in CORS site.

---

I don't think that we should show a screen with neither "open" nor "closed" selected, even if the model says it's null. The default is "open", even if the database says it's null. We can check tomorrow in the stand-up.

---

OK, I'll set "PUBLIC" as the default value for siteStatus radio buttons.